### PR TITLE
Replace ops with compiles decorator by regular registry

### PR DESCRIPTION
**Pull request by datapythonista**
Having operations defined in two places make things more complex for no reason. Removing the `compiles` decorator for most backends, and using the regular registry instead.

`pyspark` seems to have mostly all operations defined using `compiles`. I think it'd make sense to have a registry like the other backends. And that would help if we move towards #2659. But will open an issue for that, and address it separately.

### CODE REVIEW

**Comment by datapythonista** [comment]
This is blocked by #2727. Once the SQLAlchemy backend code is refactored, we need to move the ops there to its `registry.py` file, and remove the `compiles` decorator.

Also, there are couple of examples in the tutorial using the `compiles` decorator that need to be updated.

**Comment by jreback** [comment]
can you merge master and let the ci run again here

File: ibis/backends/base/sql/alchemy/translator.py [hunk 1]
```diff
@@ -82,8 +77,18 @@ def _nullifzero(expr):
...
 
-@compiles(ops.Divide)
+# TODO This was previously implemented with the legacy `@compiles` decorator.
+# This definition should now be in the registry, but there is some magic going
+# on that things fail if it's not defined here (and in the registry
+# `operator.truediv` is used.
 def _true_divide(t, expr):
+    # Having imports here, so this is self-contained and faster to clean up
+    # when this function is moved to the registry
+    import ibis.expr.types as ir
```
Comment (jreback) [inline ibis/backends/base/sql/alchemy/translator.py:87]: this doesn't make sense, these are standard ibis imports

**Review by jreback — CHANGES_REQUESTED** [review]
rebase as well

**Comment by datapythonista** [comment]
Rebased, still all green. Thanks @jreback 

I also moved the imports to the top. Agree that they belong here, but should be able to delete that function soon, so left them there to remove it quicker with all its imports. Not a big deal.

**Comment by jreback** [comment]
> Rebased, still all green. Thanks @jreback
> 
> I also moved the imports to the top. Agree that they belong here, but should be able to delete that function soon, so left them there to remove it quicker with all its imports. Not a big deal.

thanks, yeah certainly removing that function is fine (followup)

**Review by jreback — APPROVED** [review]

**Pull request merged**